WooCommerce/Classes/POS/Presentation/CardReaderConnection/CardReaderConnectionStatusView.swift

```diff
@@ -52,6 +54,22 @@ struct CardReaderConnectionStatusView: View {
         }
         .font(Constants.font)
     }
+
+    private var connectedFontColor: Color {
+        if totalsViewModel.paymentState == .processingPayment {
+            return .posSecondaryTextDark
+        } else {
+            return .primaryText
+        }
+    }
+
+    private var disconnectedFontColor: Color {
+        if totalsViewModel.paymentState == .processingPayment {
+            return .posSecondaryTextDark
+        } else {
+            return Color(.wooCommercePurple(.shade60))
+        }
+    }
```

Comment:
I think that needing this unrelated viewModel to change the colour indicates tighter coupling than we really want here...

Perhaps the view's colour should be set with a view modifier external to the view...

Reply:
Yes, I agree. This all was a last-minute addition to the PR, likely poorly thought out. I'll check out the alternatives you suggested, thanks!